src/components/Reports/AdditionalSalaryRequest/Shared/AdditionalSalaryRequestContext.tsx

```diff
@@ -59,10 +69,94 @@ export const AdditionalSalaryRequestProvider: React.FC<Props> = ({
   const [sectionIndex, setSectionIndex] = useState(0);
   const selectedSection = sectionOrder[sectionIndex];
   const [isDrawerOpen, setIsDrawerOpen] = useState(true);
+
   const toggleDrawer = useCallback(() => {
     setIsDrawerOpen((prev) => !prev);
   }, []);
 
+  const createCurrencyValidation = useCallback(
+    (fieldName: string, max?: number) => {
+      let schema = amount(fieldName, t);
+      if (max) {
+        schema = schema.max(
+          max,
+          t('Exceeds ${{amount}} limit', {
+            amount: currencyFormat(max, 'USD', locale, {
+              showTrailingZeros: true,
+            }),
+          }),
+        );
+      }
+      return schema;
+    },
+    [t],
+  );
+
+  const initialValues: CompleteFormValues = useMemo(
```

Comment:
optional: I don't think this needs to be memoized if you don't pass `enableReinitialize: true` on `useFormik.